File: Code/C++/1109. Corporate Flight Bookings.cpp
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> res(n+1);
        for (auto& booking:bookings){
            int first = booking[0]-1, second = booking[1], k = booking[2];
            res[first] += k;
            res[second] -= k;
        }
        for (int i=1;i<=n;i++){
            res[i] += res[i-1];
        }
        res.pop_back();
        return res;

    }
};

File: Code/Python/1109. Corporate Flight Bookings.py
from itertools import accumulate

class Solution:
    def corpFlightBookings(self, bookings, n):
        res = [0]*(n+1)
        for start, end, k in bookings: 
            res[start-1] += k # start from 1
            res[end] += k
        return list(accumulate(res[:-1]))